File: Server/API/Controllers/UserController.cs
using API.DTO;
using AutoMapper;
using DataAccess.Models;
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace API.Controllers
{
    [Route("api/user")]
    public class UserController : ControllerBase
    {
        private readonly MediaContext _context;
        private readonly IMapper _mapper;
        public UserController(MediaContext context, IMapper mapper)
        {
            _context = context;
            _mapper = mapper;
        }

        [Route("settings")]
        [HttpPost]
        public async Task<ActionResult> UpdateUserSettings([FromBody] UpdateUserSettings updateUserSettings)
        {
            foreach (var library in updateUserSettings.ExistingLibraries.Concat(updateUserSettings.NewLibraries))
            {
                if (!Directory.Exists(library.FolderPath))
                {
                    return BadRequest("The specified folder path is invalid");
                }
            }
            foreach (var library in updateUserSettings.ExistingLibraries.Concat(updateUserSettings.NewLibraries))
            {
                var existing = _context.Libraries.FirstOrDefault(x => x.Name.ToUpper() == library.Name.ToUpper());
                if (existing?.Id != library.Id)
                {
                    return BadRequest($"A library called {library.Name} already exists");
                }
            }
            foreach (var library in updateUserSettings.ExistingLibraries)
            {
                var updateLibrary = _context.Libraries.FirstOrDefault(x => x.Id == library.Id);
                if (updateLibrary != null)
                {
                    updateLibrary.FolderPath = library.FolderPath;
                    updateLibrary.Name = library.Name;
                }
            }
            foreach (var library in updateUserSettings.NewLibraries)
            {
                var newLibrary = new Library()
                {
                    Name = library.Name,
                    CanRemove = true,
                    FolderPath = library.FolderPath
                };
                _context.Libraries.Add(newLibrary);
            }
            foreach (var library in updateUserSettings.RemovedLibraries)
            {
                var removeLibrary = _context.Libraries.FirstOrDefault(x => x.Id == library.Id);
                var mediaItems = _context.MediaItems.Where(x => x.LibraryId == library.Id);
                _context.MediaItems.RemoveRange(mediaItems);
                _context.Libraries.Remove(removeLibrary);
            }
            await _context.SaveChangesAsync();
            return Ok();
        }

        [Route("settings")]
        [HttpGet]
        public UserSettingsDTO GetUserSettings()
        {
            var userSettings = _context.UserSettings.FirstOrDefault();
            var userSettingsDTO = new UserSettingsDTO(
                userSettings.Id,
                _mapper.Map<List<LibraryDTO>>(_context.Libraries));
            return userSettingsDTO;
        }
    }
}

File: go-server/controllers/userController.go
package controllers

import (
	"errors"
	"go-server/database"
	"os"
)

func GetUserSettings() UserSettings {
	db := database.GetDatabase()
	var userSettingsEntity database.UserSettings
	db.First(&userSettingsEntity)
	var libraries []database.Library
	db.Find(&libraries)
	var librariesDto []Library
	for _, library := range libraries {
		librariesDto = append(librariesDto, Library{Id: library.Id, Name: library.Name,
			CanRemove: library.CanRemove, FolderPath: library.FolderPath})
	}
	userSettings := UserSettings{
		Id:        userSettingsEntity.Id,
		Libraries: librariesDto,
	}
	return userSettings
}

func UpdateSettings(updateUserSettings UpdateUserSettings) error {
	for _, library := range append(updateUserSettings.ExistingLibraries, updateUserSettings.NewLibraries...) {
		_, err := os.Stat(library.FolderPath)
		if err == nil {
			continue
		}
		if os.IsNotExist(err) {
			return errors.New("The specified folder path is invalid")
		}
	}

	db := database.GetDatabase()
	for _, library := range append(updateUserSettings.ExistingLibraries, updateUserSettings.NewLibraries...) {
		var libraryEntity database.Library
		result := db.Where(&Library{Name: library.Name}).First(&libraryEntity)
		if result.Error != nil {
			libraryEntity.FolderPath = library.FolderPath
			libraryEntity.Name = library.Name
			db.Save(&libraryEntity)
		} else {
			return result.Error
		}
	}

	addedLibraries := []database.Library{}
	for _, library := range updateUserSettings.NewLibraries {
		addedLibraries = append(addedLibraries, database.Library{Name: library.Name, FolderPath: library.FolderPath, CanRemove: true})
	}
	db.Create(&addedLibraries)

	for _, library := range updateUserSettings.RemovedLibraries {
		db.Where(&database.MediaItem{LibraryId: library.Id}).Delete(database.MediaItem{})
		db.Delete(&database.Library{}, library.Id)
	}
	return nil
}
